### ISSUE TITLE
site specific feature flags loading after refresh

### ISSUE THREAD

**Issue by dextel2** [splitio/react-client#148]
I have a Context in react for featureFlags which loads feature flags for specific sites or domains. Currently I have to switch domains in order to load that domains featureFlag. Here's a code snippet 

```ts
const ProvideFeatureFlagsRaw = ({ children }: IProvideFeatureFlagsProps) => {
  const featureFlagsContextValue = useFeatureFlagsProvider();
  return (
    <FeatureFlagsContext.Provider value={featureFlagsContextValue}>
      {children}
    </FeatureFlagsContext.Provider>
  );
};

const SDK_CONFIG = {
  core: {
    authorizationKey: process.env.REACT_APP_SPLIT_API_KEY,
    key: cookie.getJSON(TENANT_COOKIE_NAME)?.domain,
  },
  debug: false,
} as SplitIO.IBrowserSettings;

const withSplit = withSplitFactory(SDK_CONFIG);

const ProvideFeatureFlags = withSplit(ProvideFeatureFlagsRaw);

export default ProvideFeatureFlags;
```

The `useFeatureFlagsProvider()` basically is hook which returns all the feature flags with the state either they are on or off

**Comment by hbqdev** [comment]
hi @dextel2 
Can you please clarify what is the exact issue you are having?

Regards

**Comment by dextel2** [comment]
hi @hbqdev 

The issue I am facing is that when I am switching from domain _a_ to domain _b_ my SDK_COFIG is not getting updated with key of domain _b_ while being in domain _a_ I still have the context of _b_, How can I re-init my SDK_CONFIG ?

**Comment by hbqdev** [comment]
Hi @dextel2 
Are you referring to your environment? If you wish to use a different environment you will need to change the authorizationKey to use the API key from that environment.

The key `    key: cookie.getJSON(TENANT_COOKIE_NAME)?.domain,` is the key you pass in for treatment evaluation, and it has nothing to do with any domain switching. The key will be evaluated and if there is a matching treatment, you will get the treatment back.

**Comment by hbqdev** [comment]
hi @dextel2 
It has been a while since we heard back from you.
I'm closing out this ticket. If there's still an open issue or if we can be of further assistance on this topic, you can send an email to support@split.io and we will be happy to assist.

Regrds

**Comment by dextel2** [comment]
hi @hbqdev sorry I got drifted in work I'll update this thread

**Comment by dextel2** [comment]
So we are using _tenant_ specific system i.e. there will always one domain name but the tenant will be dynamic for example 
![image](https://github.com/splitio/react-client/assets/16697307/6298af44-94a5-4b45-b34b-326b10cb27bc)
in this calse `tenant.domain` will be shoe but when I switch from shoe to lotus the reference to the site feature flag is not being updated unless we refresh the page

**Comment by EmilianoSanchez** [comment]
Hi @dextel2 , 

Unfortunately, you cannot update the underlying SDK config (and factory) inside the `SplitFactory` component or the `withSplitFactory` HOC (which internally uses `SplitFactory`).

As explained here https://github.com/splitio/react-client/blob/master/types/SplitFactory.d.ts#L8, the underlying SDK factory is set on the constructor, and cannot be changed during the component lifecycle, even if the component is updated with a different config prop, as I understand you are attempting to do.

But you can use `SplitClient` or `withSplitClient` if you want to switch the user key from `shoe` to `lotus` dynamically and then update the feature flag evaluation. 

For example, you can update your original code snippet [here](https://github.com/splitio/react-client/issues/148#issue-1780343403) as follows: 

```
const ProvideFeatureFlagsRaw = ({ children }: IProvideFeatureFlagsProps) => {
  const featureFlagsContextValue = useFeatureFlagsProvider();
  return (
    <FeatureFlagsContext.Provider value={featureFlagsContextValue}>
      {children}
    </FeatureFlagsContext.Provider>
  );
};

const SDK_CONFIG = {
  core: {
    authorizationKey: process.env.REACT_APP_SPLIT_API_KEY,
    key: cookie.getJSON(TENANT_COOKIE_NAME)?.domain,
  },
  debug: false,
} as SplitIO.IBrowserSettings;

const ProvideFeatureFlags = ({ domain }) => {
  return (
    <SplitFactory config={SDK_CONFIG} >
      <SplitClient splitKey={domain || SDK_CONFIG.core.key} >
        <ProvideFeatureFlagsRaw />
      </SplitClient>
    </SplitFactory>
  )  
);

export default ProvideFeatureFlags;
```

In this example, you can dynamically change the `domain` prop of the ProvideFeatureFlags component, and the feature flag evaluations inside `ProvideFeatureFlagsRaw` will "react" to changes in the `domain` key.

**Comment by dextel2** [comment]
Hi @EmilianoSanchez I appreciate your effort for the above explanation, but as soon I switch from `shoe` to `lotus` it auth method from split sdk does not re-inits but on refresh it does. FYR this is made from the the react context API and is being used as a Provider.

![image](https://github.com/splitio/react-client/assets/16697307/f481ed21-523f-4006-b2cd-a32e2a029334)

**Comment by EmilianoSanchez** [comment]
Hi @dextel2 
I am sorry you weren't able to solve the issue yet. 
Could you provide more details about the implementation of your `useFeatureFlagsProvider` hook and the `ProvideSiteFeatureFlags` component in the screenshot? like, for example, what modules from the React SDK library are you calling and how?

Also, you can contact [support@split.io](mailto:support@split.io) if you prefer to share your code with us and continue from there.

**Comment by EmilianoSanchez** [comment]
Hi @dextel2 , 

It has been a while since your last message, but I wanted to inform you that we have released React SDK v1.10.1.
This version includes a bug fix for the `SplitClient` component that might be related to your issue.

Basically, there was a time window between the "render" and "commit" phases of the React component's lifecycle where, if an SDK event was emitted, the `SplitClient` component was not updating (i.e., re-rendering).
